imagebuildah/build.go

```diff
@@ -879,22 +879,46 @@ func (s *StageExecutor) Execute(ctx context.Context, stage imagebuilder.Stage, b
 	}
 	children := stage.Node.Children
 
+	// A helper function to only log "COMMIT" as an explicit step if it's
+	// the very last step of a (possibly multi-stage) build.
+	logCommit := func(output string, instruction int) {
+		if instruction < len(children)-1 || s.index < s.stages-1 {
+			return
+		}
+		if output != "" {
+			logrus.Debugf("COMMIT %q", output)
+			if !s.executor.quiet {
+				s.executor.log("COMMIT %s", output)
+			}
+		} else {
+			logrus.Debugf("COMMIT")
+			if !s.executor.quiet {
+				s.executor.log("COMMIT")
+			}
+		}
+	}
+
 	if len(children) == 0 {
 		// There are no steps.
 		if s.builder.FromImageID == "" || s.executor.squash {
 			// We either don't have a base image, or we need to
 			// squash the contents of the base image.  Whichever is
 			// the case, we need to commit() to create a new image.
+			logCommit(s.output, -1)
 			if imgID, ref, err = s.commit(ctx, ib, getCreatedBy(nil), s.output); err != nil {
 				return "", nil, errors.Wrapf(err, "error committing base container")
 			}
 		} else {
 			// We don't need to squash the base image, so just
 			// reuse the base image.
+			logCommit(s.output, -1)
 			if imgID, ref, err = s.copyExistingImage(ctx, s.builder.FromImageID, s.output); err != nil {
 				return "", nil, err
 			}
 		}
+		if s.executor.iidfile == "" {
+			fmt.Fprintf(s.executor.out, "--> %s\n", imgID)
+		}
```

Comment:
Consideration.  Would it make sense to move this if statement to a small function?  Just on the off chance we ever need to adjust it slightly, we'd only have to do it in one spot.

Reply:
Might as well.